Deployment/Jenkins/agent/agent-instance.groovy

```diff
@@ -0,0 +1,258 @@
+                                    "ParameterValue": "${env.SubnetId}"
+                                },
+                                {
+                                    "ParameterKey": "ZapAgentUrl",
+                                    "ParameterValue": "${env.ZapAgentUrl}"
+                                }
+                            ]
+                        /
+                }
+            }
+        }
+        stage ('Prepare AWS Environment') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to delete any active ${CfnStackRoot} stacks... "
+                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
+
+                        aws cloudformation wait stack-delete-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
+                    '''
+                }
+            }
+        }
+        stage ('Launch Jenkins Master Stack') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash -xe
+                        printf 'export SIGNED_URL="%s"' "\$( aws s3 presign ${ChainScriptUrl} )" > /tmp/SIGNED_URL.txt
+                        source /tmp/SIGNED_URL.txt
+                        printenv | sort
+                        echo "Attempting to create stack ${CfnStackRoot}..."
+                        aws --region "${AwsRegion}" cloudformation create-stack --stack-name "${CfnStackRoot}" \
+                          --disable-rollback --capabilities CAPABILITY_NAMED_IAM \
+                          --template-url "${TemplateUrl}" \
+                          --parameters file://<( sed "s#__SIGNED_URL__#${SIGNED_URL//&/\\&}#" agent.instance.parms.json )
+
+                        aws cloudformation wait stack-create-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
```

Comment:
I really, really, _really_ don't like twiddling my thumbs as minutes go by without output (but won't reject over this).

Reply:
We can always change it back down the road.